File: deal/src/main/java/ru/artemev/deal/dto/LoanOfferDTO.java
package ru.artemev.deal.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;

@Data
@Builder
@Schema(description = "Предложение по кредиту")
@JsonDeserialize(builder = LoanOfferDTO.LoanOfferDTOBuilder.class)
public class LoanOfferDTO {
  @Schema(description = "Итоговая сумма", example = "15000")
  @JsonProperty("totalAmount")
  private BigDecimal totalAmount;

  @Schema(description = "Срок займа", example = "10")
  @JsonProperty("term")
  private Integer term;

  @Schema(description = "Ежемесячный платеж", example = "1500")
  @JsonProperty("monthlyPayment")
  private BigDecimal monthlyPayment;

  @Schema(description = "Процентная ставка", example = "10")
  @JsonProperty("rate")
  private BigDecimal rate;

  @Schema(description = "Страховка", example = "true")
  @JsonProperty("isInsuranceEnabled")
  private Boolean isInsuranceEnabled;

  @Schema(description = "Зарплатный клиент", example = "true")
  @JsonProperty("isSalaryClient")
  private Boolean isSalaryClient;

  @Schema(description = "Номер предложения", example = "4")
  @JsonProperty("applicationId")
  private Long applicationId;

  @Schema(description = "Запрашиваемая сумма", example = "100000")
  @JsonProperty("requestedAmount")
  private BigDecimal requestedAmount;
}

File: deal/src/main/kotlin/ru/kit/deal/dto/LoanOfferDTO.kt
package ru.kit.deal.dto

import io.swagger.v3.oas.annotations.media.Schema
import java.math.BigDecimal

@Schema(description = "Предложение по кредиту")
data class LoanOfferDTO(
    @Schema(description = "Номер предложения", example = "4")
    private val applicationId: Long? = null,

    @Schema(description = "Запрашиваемая сумма", example = "100000")
    private val requestedAmount: BigDecimal? = null,

    @Schema(description = "Итоговая сумма", example = "15000")
    private val totalAmount: BigDecimal? = null,

    @Schema(description = "Срок займа", example = "10")
    private val term: Int? = null,

    @Schema(description = "Ежемесячный платеж", example = "1500")
    private val monthlyPayment: BigDecimal? = null,

    @Schema(description = "Процентная ставка", example = "10")
    private val rate: BigDecimal? = null,

    @Schema(description = "Страховка", example = "true")
    private val isInsuranceEnabled: Boolean = false,

    @Schema(description = "Зарплатный клиент", example = "true")
    private val isSalaryClient: Boolean = false,
)